vespaclient-java/src/main/java/com/yahoo/vespasignificance/SignificanceModelGenerator.java

```diff
@@ -0,0 +1,147 @@
+// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+package com.yahoo.vespasignificance;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.yahoo.document.DataType;
+import com.yahoo.document.Document;
+import com.yahoo.document.DocumentPut;
+import com.yahoo.document.DocumentType;
+import com.yahoo.document.DocumentTypeManager;
+import com.yahoo.document.Field;
+import com.yahoo.document.datatypes.FieldValue;
+import com.yahoo.document.json.DocumentOperationType;
+import com.yahoo.document.json.JsonReader;
+import com.yahoo.document.json.ParsedDocumentOperation;
+import com.yahoo.language.Language;
+import com.yahoo.language.opennlp.OpenNlpLinguistics;
+import com.yahoo.language.process.StemMode;
+import com.yahoo.language.process.Token;
+import com.yahoo.language.process.TokenScript;
+import com.yahoo.language.process.TokenType;
+import com.yahoo.language.process.Tokenizer;
+import com.yahoo.language.significance.impl.DocumentFrequencyFile;
+import com.yahoo.language.significance.impl.SignificanceModelFile;
+import com.yahoo.text.Utf8;
+
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * @author MariusArhaug
+ */
+public class SignificanceModelGenerator {
+
+    private final ClientParameters clientParameters;
+    private final Tokenizer tokenizer;
+    private final HashMap<String, Long> documentFrequency = new HashMap<>();
+    private final Language language;
+    private final ObjectMapper objectMapper;
+    private final static JsonFactory parserFactory = new JsonFactory();
+
+    final DocumentTypeManager types = new DocumentTypeManager();
+    final DocumentType docType;
+    private final static String VERSION = "1.0";
+    private final static String ID = "1";
+    private final static String SIGNIFICANCE_DESCRIPTION = "Significance model for input file";
+    private final static String DOC_FREQ_DESCRIPTION = "Document frequency for language";
+
+    public SignificanceModelGenerator(ClientParameters clientParameters) {
+        this.clientParameters = clientParameters;
+        OpenNlpLinguistics openNlpLinguistics = new OpenNlpLinguistics();
+        tokenizer = openNlpLinguistics.getTokenizer();
+        objectMapper = new ObjectMapper();
+
+        language = Language.fromLanguageTag(clientParameters.language);
+
+        docType = new DocumentType(language.languageCode().toLowerCase());
```

Comment:
Add document type as a new CLI parameter instead of assuming it's named after the language.

Reply:
Added docType argument to cli       options.addOption(Option.builder("d")
                .hasArg(true)
                .desc("Document type identifier")
                .longOpt(DOC_TYPE_OPTION)
                .build());